### PR TITLE
Add a few more things to the helper python scripts.

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Add ability to run an onnx format test dir from python.
Add documentation with usage examples.
Cleanup some miscellaneous things, mainly pep8 related.

**Motivation and Context**
Added some things when investigating github issues where just a model was provided. 
Figured documentation with example usage would help others utilize the scripts when doing similar investigations in the future.

### CODE REVIEW

File: tools/python/PythonTools.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+In order to import ort_test_dir_utils you need to either 
+  - run python from the `<onnxruntime root dir`>\tools\python directory
+  - add the directory to your PYTHONPATH 
+  - add the directory to sys.path prior to importing
+
+e.g. to add to sys.path
+```python 
+import sys
+sys.path.append('<onnxruntime root dir>\tools\python')
```
Comment (xadupre) [inline tools/python/PythonTools.md:15]: / is better and it works on windows

File: tools/python/PythonTools.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Example usage:
+
+```python
+import sys
+import ort_test_dir_utils
+import onnx_test_data_utils
+
+# example model with two float32 inputs called 'input1' (dims: {2, 1}) and 'input2' (dims: {'dynamic', 4})
+model_path = r'<onnxruntime root dir>\onnxruntime\test\testdata\transform\expand_elimination.onnx'
```
Comment (xadupre) [inline tools/python/PythonTools.md:55]: /

File: tools/python/PythonTools.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+# when using the default data generation any symbolic dimension values must be provided
+symbolic_vals = {'dynamic':2} # provide value for symbolic dim named 'dynamic' in 'input2'
+
+# let create_test_dir create random input in the (arbitrary) default range of -10 to 10. 
+# it will create data of the correct type based on the model.
+ort_test_dir_utils.create_test_dir(model_path, r'd:\temp\examples', 'test1', symbolic_dim_values_map=symbolic_vals)
```
Comment (xadupre) [inline tools/python/PythonTools.md:62]: I would use a relative path instead of d:/temp.

**Comment by skottmckay** [comment]
> By using `os.path.join('path1', os.sep, 'path2', os.sep, 'path3')` this code would be portable for both windows and linux.
> Also, using a relative path from the could prevent the weird `os.path.join('c:', 'path1', 'path2')` scenario too

run_test_dir uses os.path.join when just a directory name is provided.
`        models = glob.glob(os.path.join(model_dir, '*.onnx'))`

That seems to work fine on linux for absolute and relative paths, so I'm wondering when explicitly specifying the separator is required. 

```
>>> ort_test_dir_utils.run_test_dir('/home/scott/Downloads/ort_perf_test_models/matmul2d_float')
Running tests in /home/scott/Downloads/ort_perf_test_models/matmul2d_float
/home/scott/Downloads/ort_perf_test_models/matmul2d_float/test_data_set_4
PASS
>>> ort_test_dir_utils.run_test_dir('../../../../../Downloads/ort_perf_test_models/matmul2d_float')
Running tests in /home/scott/Downloads/ort_perf_test_models/matmul2d_float
/home/scott/Downloads/ort_perf_test_models/matmul2d_float/test_data_set_4
PASS

```

**Comment by thiagocrepaldi** [comment]
> 
> 
> > By using `os.path.join('path1', os.sep, 'path2', os.sep, 'path3')` this code would be portable for both windows and linux.
> > Also, using a relative path from the could prevent the weird `os.path.join('c:', 'path1', 'path2')` scenario too
> 
> run_test_dir uses os.path.join when just a directory name is provided.
> ` models = glob.glob(os.path.join(model_dir, '*.onnx'))`
> 
> That seems to work fine on linux for absolute and relative paths, so I'm wondering when explicitly specifying the separator is required.
> 
> ```
> >>> ort_test_dir_utils.run_test_dir('/home/scott/Downloads/ort_perf_test_models/matmul2d_float')
> Running tests in /home/scott/Downloads/ort_perf_test_models/matmul2d_float
> /home/scott/Downloads/ort_perf_test_models/matmul2d_float/test_data_set_4
> PASS
> >>> ort_test_dir_utils.run_test_dir('../../../../../Downloads/ort_perf_test_models/matmul2d_float')
> Running tests in /home/scott/Downloads/ort_perf_test_models/matmul2d_float
> /home/scott/Downloads/ort_perf_test_models/matmul2d_float/test_data_set_4
> PASS
> ```

`os.sep` is not strictly necessary, especially for newer versions of Python. Although it is the most correct syntactically speaking, you can get away without it too :)

File: tools/python/PythonTools.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+Example usage:
+
+```python
+import sys
+import ort_test_dir_utils
+
+try:
+    ort_test_dir_utils.run_test_dir('temp/examples/test1')
```
Comment (thiagocrepaldi) [inline tools/python/PythonTools.md:99]: Some of the document instructions include non portable paths too `os.path.join('temp','examples','test1')`

Comment (skottmckay) [reply tools/python/PythonTools.md:99]: I believe Windows accepts either forward or backwards slashes.

```
D:\src\github\onnxruntime> python
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.chdir('/src/github')
>>> os.getcwd()
'D:\\src\\github'
```

**Comment by skottmckay** [comment]
> `os.sep` is not strictly necessary, especially for newer versions of Python. Although it is the most correct syntactically speaking, you can get away without it too :)

I was under the impression that using os.path.join was to avoid having to do the os.sep manually. The documentation for python 2.7 says it inserts os.sep as needed. I don't claim to be a python expert though so would be interested in reading the official documentation on the best practices for handling paths. Can you point me to it?

> os.path.join(path, *paths)
> Join one or more path components intelligently. The return value is the concatenation of path and any members of *paths with exactly one directory separator (os.sep) following each non-empty part except the last, meaning that the result will only end in a separator if the last part is empty. If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

**Comment by thiagocrepaldi** [comment]
> 
> 
> > `os.sep` is not strictly necessary, especially for newer versions of Python. Although it is the most correct syntactically speaking, you can get away without it too :)
> 
> I was under the impression that using os.path.join was to avoid having to do the os.sep manually. The documentation for python 2.7 says it inserts os.sep as needed. I don't claim to be a python expert though so would be interested in reading the official documentation on the best practices for handling paths. Can you point me to it?
> 
> > os.path.join(path, *paths)
> > Join one or more path components intelligently. The return value is the concatenation of path and any members of *paths with exactly one directory separator (os.sep) following each non-empty part except the last, meaning that the result will only end in a separator if the last part is empty. If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

https://docs.python.org/3/library/os.path.html#os.path.join

**Pull request merged**